docs/src/api/commandline.md

```diff
@@ -0,0 +1,221 @@
+`uniffi-bindgen-react-native` the command line utility that ties together much of the building of Rust, and the generating the bindings and turbo-modules.
+
+Most commands take a `--config FILE` option. This is a YAML file which collects commonly used options together, and is [documented here](config-yaml.md).
+
+The command can be invoked with `./node_modules/.bin/uniffi-bindgen-react-native`.
+
+Running the following command from within your project:
+
+```sh
+alias ubrn=$(pwd)/node_modules/.bin/uniffi-bindgen-react-native
+```
+allows you to run the command as `ubrn`, which is simpler to type. From hereon, commands will be given as `ubrn` commands.
+
+# `ubrn` Commands
+
+Running `ubrn --help` gives the following output:
+
+```sh
+Usage: uniffi-bindgen-react-native <COMMAND>
+
+Commands:
+  checkout  Checkout a given Github repo into `rust_modules`
+  build     Build (and optionally generate code) for Android or iOS
+  generate  Generate bindings or the turbo-module glue code from the Rust
+  help      Print this message or the help of the given subcommand(s)
+
+Options:
+  -h, --help  Print help
+```
+
+## `checkout`
+Checkout a given Git repo into `rust_modules`.
+
+```sh
+Usage: uniffi-bindgen-react-native checkout [OPTIONS] <REPO>
+
+Arguments:
+  <REPO>  The repository where to get the crate
+
+Options:
+      --config <CONFIG>
+      --branch <BRANCH>  The branch or tag which to checkout [default: main]
+  -h, --help             Print help
+```
+The checkout command can be operated in two ways, either:
+1. with a `REPO` argument and optional `--branch` argument. OR
+2. with a config file which may specify a repo and branch, or just a `directory`.
+
+## `build android`
+
+Build the crate for use on an Android device or emulator, using `cargo ndk`, which in turn uses Android Native Development Kit.
```

Comment:
One thing that we had to do here, was make sure that we were linking to the correct NDK. 

This changes from RN version to version, but in our usage we had to set an ANDROID_NDK_HOME variable in our script for this to pick up the appropriate version.

You can find the version you need in your react-native android/build.gradle file in the `ndkVersion` variable.

export ANDROID_NDK_HOME=${ANDROID_SDK_ROOT}/ndk/26.1.10909125/